src/lib.rs

```diff
@@ -565,35 +778,86 @@ impl<T: Data + 'static> WinHandler for UiMain<T> {
 }
 
 impl BaseState {
+    /// The "hot" (aka hover) status of a widget.
+    ///
+    /// A widget is "hot" when the mouse is hovered over it. Widgets will
+    /// often change their appearance as a visual indication that they
+    /// will respond to mouse interaction.
+    ///
+    /// The hot status is computed from the widget's layout rect. In a
+    /// container hierarchy, all widgets with layout rects containing the
+    /// mouse position have hot status.
+    ///
+    /// Discussion: there is currently some confusion about whether a
+    /// widget can be considered hot when some other widget is active (for
+    /// example, when clicking to one widget and dragging to the next).
+    /// The documentation should clearly state the resolution.
     pub fn is_hot(&self) -> bool {
         self.is_hot
     }
 
+    /// The active status of a widget.
+    ///
+    /// Active status generally corresponds to a mouse button down. Widgets
+    /// with behavior similar to a button will call [`set_active`] on mouse
+    /// down and then up.
+    ///
+    /// When a widget is active, it gets mouse events even when the mouse
+    /// is dragged away.
+    ///
+    /// [`set_active`]: struct.EventCtx.html#method.set_active
     pub fn is_active(&self) -> bool {
         self.is_active
     }
 
+    /// The focus status of a widget.
+    ///
+    /// A widget can request focus using the [`request_focus`] method.
+    /// This will generally result in a separate event propagation of
+    /// a `FocusChanged` method, including sending `false` to the previous
+    /// widget that held focus.
+    ///
+    /// Only one leaf widget at a time has focus. However, in a container
+    /// hierarchy, all ancestors of that leaf widget are also invoked with
+    /// `FocusChanged(true)`.
+    ///
+    /// Discussion question: is "is_focused" a better name?
```

Comment:
As with `is_hot` and `is_active`, this docstring should concisely state what "focus" means.